File: coding/H3_Merge_K_Sorted_Arrays/Solution/Solution.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        if (!sc.hasNextInt()) return;
        int K = sc.nextInt();

        int[] len = new int[K];
        int total = 0;

        // Read sizes
        for (int i = 0; i < K; i++) {
            len[i] = sc.nextInt();
            total += len[i];
        }

        int[] all = new int[total];
        int idx = 0;

        // Read elements
        for (int i = 0; i < K; i++) {
            for (int j = 0; j < len[i]; j++) {
                all[idx++] = sc.nextInt();
            }
        }

        // Sort
        Arrays.sort(all);

        // Print
        for (int i = 0; i < total; i++) {
            System.out.print(all[i] + " ");
        }

        sc.close();
    }
}

File: coding/H3_Merge_K_Sorted_Arrays/Solution/Solution.c
#include <stdio.h>
#include <stdlib.h>

int cmp(const void* a, const void* b){
    return (*(int*)a - *(int*)b);
}

int main(){
    int K;
    if(scanf("%d",&K)!=1) return 0;

    int total = 0;
    int *len = malloc(K*sizeof(int));

    // First read sizes
    for(int i=0;i<K;i++){
        scanf("%d",&len[i]);
        total += len[i];
    }

    int *all = malloc(total * sizeof(int));
    int idx = 0;

    // Now read elements
    for(int i=0;i<K;i++){
        for(int j=0;j<len[i];j++){
            scanf("%d",&all[idx++]);
        }
    }

    // Sort all elements
    qsort(all, total, sizeof(int), cmp);

    // Print merged sorted array
    for(int i=0;i<total;i++){
        printf("%d ", all[i]);
    }

    return 0;
}
